### PR TITLE
fix miscellaneous bugs related to new resource hierarchy migration

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
What does this implement/fix? Explain your changes.
---------------------------------------------------
fix miscellaneous bugs related to new resource hierarchy migration

Does this close any currently open issues?
------------------------------------------
[AB#1931521](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931521): [Test] Many error messages include meaningless string 'null'
[AB#1931531](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931531): [Test] Occurred an error when refreshing favorite list including deleted resources
[AB#1931523](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931523): [Test] Add web app setting, no progress and event log display, cannot save normally.
[AB#1931889](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931889): [Test] Should disable Straming Logs/Profile Flight Recorder/Edit Deployment for deleted function/webapp/deployment in favorite list.
[AB#1931884](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931884): [Test] NPE when refreshing favorite list for a resource which isn't in the subscription
[AB#1931880](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1931880): [Test] NPE when showing properties or starting streaming logs in favorite for a resource which isn't in the subscription
[AB#1928238](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1928238): [Test] Spring cloud apps keep loading if expand the service at first time

Any relevant logs, screenshots, error output, etc.?
-------------------------------------
N/P

Any other comments?
-------------------
N/P

Has this been tested?
---------------------------
- [x] Tested

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/AbstractAzResourceModule.java [hunk 1]
```diff
@@ -201,7 +201,10 @@ public T create(@Nonnull AzResource.Draft<T, R> draft) {
     @Nonnull
     <D extends AzResource.Draft<T, R>> D update(@Nonnull T resource) {
         log.debug("[{}]:update(resource:{})", this.name, resource);
-        if (resource instanceof AzResource.Draft) {
+        if (resource.isDraftForCreating()) {
+            log.warn("[{}]:updating(resource:{}) from a draftForCreating", this.name, resource);
+        }
+        if (resource.isDraft()) {
             return this.cast(resource);
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/AbstractAzResourceModule.java]: should not allow update a draft.

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/AzResource.java [hunk 2]
```diff
@@ -164,13 +164,10 @@ public int hashCode() {
 
         default T commit() {
             synchronized (this) {
-                if (this.isCommitted()) {
-                    throw new AzureToolkitRuntimeException("this draft has been committed.");
-                }
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/AzResource.java:169]: should not allow committing a draft more than once.

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/AzResourceBase.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ default FormalStatus getFormalStatus() {
         private static final HashSet<String> writingStatus = Sets.newHashSet("writing", "pending", "processing", "creating", "updating", "deleting",
             "starting", "stopping", "restarting", "scaling");
         private static final HashSet<String> readingStatus = Sets.newHashSet("reading", "loading", "refreshing");
-        private static final HashSet<String> deletedStatus = Sets.newHashSet("deleted", "removed");
+        private static final HashSet<String> deletedStatus = Sets.newHashSet("deleted", "removed", "disconnected");
```
Comment (wangmingliang-ms) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/AzResourceBase.java:52]: remove `disconnected` and use `deleted` instead.

**Review by Flanker32 — APPROVED** [review]
LGTM

**Pull request merged**